pkg/querier/queryrange/split_by_interval.go

```diff
@@ -263,8 +271,42 @@ func forInterval(interval time.Duration, start, end time.Time, callback func(sta
 	}
 }
 
-func splitMetricByTime(r queryrangebase.Request, interval time.Duration) []queryrangebase.Request {
+// maxRangeVector returns the maximum range vector duration within a LogQL query.
+func maxRangeVector(q string) (time.Duration, error) {
+	expr, err := logql.ParseSampleExpr(q)
+	if err != nil {
+		return 0, err
+	}
+	var max time.Duration
+	expr.Walk(func(e interface{}) {
+		if r, ok := e.(*logql.LogRange); ok && r.Interval > max {
+			max = r.Interval
+		}
+	})
+	return max, nil
+}
+
+// reduceSplitIntervalForRangeVector reduce the split interval for range query based on the range vector.
+// Large range vector will be not split into smaller intervals which can cause the queries to be slow by over-processing data.
```

Comment:
```suggestion
// reduceSplitIntervalForRangeVector reduces the split interval for a range query based on the duration of the range vector.
// Large range vector durations will not be split into smaller intervals because it can cause the queries to be slow by over-processing data.
```